### PR TITLE
Tailed species now gain a negative moodlet for having their tail lost

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Species change done with maintainer authorization.

_proof

Whenever a species with a tail (felinids, lizards, and monkeys) have their tail chopped off, they get a -8 moodlet for 10 minutes and a -2 permanent moodlet.

If they get ANY tail back, the -2 permanent moodlet is removed.

If they gain the right tail for their species back, they get a -2 moodlet for a few minutes and that's that.
If they gain the WRONG tail for their species, they gain an even HEFTIER moodlet for being turned into a freak.

Because species technically lose and regain their tail on initialization, there had to be some checks added to make sure lizards and cats don't spawn depressed even though they're okay.

## Why It's Good For The Game

Like mothblocks said in the moth wing pr, enforcing policy through mechanics is neat. Lizards and Cats now have an actual reason to get upset when someone shears off their tail other than roleplay reasons.

## Changelog
:cl: Melbert
add: Tailed species now get very upset if you steal their tail from them.
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
> Like mothblocks said in the moth wing pr, enforcing policy through mechanics is neat. 

this isn't enforcing policy, this is reinforcing roleplay, but ya

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -302,15 +302,15 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 					oldorgan = null //now deleted
 			else
 				oldorgan.before_organ_replacement(neworgan)
-				oldorgan.Remove(C,TRUE)
+				oldorgan.Remove(C, TRUE, TRUE)
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/species.dm:305]: Change the booleans to be named parameters, this is unreadable.

File: code/modules/mob/living/carbon/human/species.dm [hunk 2]
```diff
@@ -336,7 +336,7 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 			if(current_organ)
 				current_organ.before_organ_replacement(replacement)
 			// organ.Insert will qdel any current organs in that slot, so we don't need to.
-			replacement.Insert(C, TRUE, FALSE)
+			replacement.Insert(C, TRUE, FALSE, TRUE)
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/species.dm:339]: Here too.

File: code/modules/mob/living/carbon/human/species_types/felinid.dm [hunk 3]
```diff
@@ -95,8 +95,8 @@
 	else
 		var/obj/item/organ/ears/cat/kitty_ears = new
 		var/obj/item/organ/tail/cat/kitty_tail = new
-		kitty_ears.Insert(H, TRUE, FALSE) //Gives nonhumans cat tail and ears
-		kitty_tail.Insert(H, TRUE, FALSE)
+		kitty_ears.Insert(H, TRUE, FALSE, TRUE) //Gives nonhumans cat tail and ears
+		kitty_tail.Insert(H, TRUE, FALSE, TRUE)
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/species_types/felinid.dm:99]: Here too.

File: code/modules/mob/living/carbon/human/species_types/lizardpeople.dm [hunk 4]
```diff
@@ -99,7 +99,7 @@
 
 		// organ.Insert will qdel any existing organs in the same slot, so
 		// we don't need to manage that.
-		new_tail.Insert(C, TRUE, FALSE)
+		new_tail.Insert(C, TRUE, FALSE, TRUE)
```
Comment (Mothblocks) [inline code/modules/mob/living/carbon/human/species_types/lizardpeople.dm:102]: And here.

File: code/modules/surgery/organs/tails.dm [hunk 5]
```diff
@@ -8,25 +8,30 @@
...
 	var/tail_type = "None"
 
-/obj/item/organ/tail/Remove(mob/living/carbon/human/H,  special = 0)
-	..()
-	if(H && H.dna && H.dna.species)
-		H.dna.species.stop_wagging_tail(H)
+/obj/item/organ/tail/Insert(mob/living/carbon/human/tail_owner, special = FALSE, drop_if_replaced = TRUE, organ_init = FALSE)
+	. = ..()
+	if(tail_owner?.dna?.species)
+		tail_owner.dna.species.on_tail_regain(tail_owner, src, organ_init)
```
Comment (Mothblocks) [inline code/modules/surgery/organs/tails.dm:14]:
```suggestion
	tail_owner?.dna?.species?.on_tail_regain(tail_owner, src, organ_init)
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by Mothblocks** [comment]
Not defines, *named arguments*.

```dm
do_something(5, foo = TRUE)
```

**Comment by RaveRadbury** [comment]
Aw man, I got permission from timber to add tripping when tails are lost, I'll see if I can get that on after this.

What about felinids that spawn without a tail? That's an option for them. Also, are you checking this for monkeys?

File: code/datums/mood_events/generic_negative_events.dm [hunk 6]
```diff
@@ -311,6 +311,25 @@
 	mood_change = -3
 	timeout = 90 SECONDS
 
+/datum/mood_event/tail_lost
+	description = "<span class='boldwarning'>My tail!! Why?!</span>\n"
+	mood_change = -8
+	timeout = 10 MINUTES
+
+/datum/mood_event/tail_balance_lost
```
Comment (RaveRadbury) [inline code/datums/mood_events/generic_negative_events.dm:319]: Yeah if you get this through I'm definitely going to PR a trip chance to this moodlet.

**Comment by MrMelbert** [comment]
i'll deal with the linting issues in the morning when I can think straighter

**Comment by bluetshirtguy** [comment]
> Aw man, I got permission from timber to add tripping when tails are lost, I'll see if I can get that on after this.
> 
> What about felinids that spawn without a tail? That's an option for them. Also, are you checking this for monkeys?

I still don't get why it'd cause a biped to trip

**Comment by MrMelbert** [comment]
> What about felinids that spawn without a tail? That's an option for them. Also, are you checking this for monkeys?

Since tail-less cats empty out their mutant organs list, they should be accounted for, but i'll double check
and yes, monkeys are checked

**Comment by RaveRadbury** [comment]
> I still don't get why it'd cause a biped to trip

Tails are used for balance. Ever met a cat with a kinked tail? They have a harder time with their acrobatics rolls. If you're used to having a counter-weight and its gone suddenly, you're probably gonna trip a bit.

**Comment by bluetshirtguy** [comment]
> > I still don't get why it'd cause a biped to trip
> 
> Tails are used for balance. Ever met a cat with a kinked tail? They have a harder time with their acrobatics rolls. If you're used to having a counter-weight and its gone suddenly, you're probably gonna trip a bit.

Cats and the vast majority of tailed animals aren't exclusively up right walking on 2 legs, gameplay wise I just see it as forcing more handicaps(like this pr) and rng(your idea on tripping) to the game.

**Comment by MrMelbert** [comment]
here I was the entire time trying to add an arg to `insert` and `remove` when I realized there was already a (poorly documented and confusingly named) argument that was exactly what I was looking for

going to do some testing tomorrow to make sure it all works fine and re-open for review then

**Comment by cacogen** [comment]
Roleplay always works best when it's in reaction to gameplay mechanics

**Comment by Sparkezel** [comment]
Dunno man, catgirls with lizard tails look better imo. Too bad they can't wag them.....

**Review by Mothblocks — APPROVED** [review]

File: code/datums/mood_events/generic_negative_events.dm (hunk 6 above)
Comment (MrMelbert) [reply code/datums/mood_events/generic_negative_events.dm:319]: if you go that route you might want to roll this into a status effect

**Pull request merged**